### PR TITLE
Instrument API classes: render_frontend_info kwargs

### PR DESCRIPTION
**Pull request by Theodlz**
adds kwargs to the render_frontend_info method of all instrument API classes. That way, reusable quantities (like unique galaxy catalog names that some instruments take as a parameter) can be computed once before we loop on each instrument and build the rjsf form for each.

### CODE REVIEW

File: skyportal/handlers/api/internal/robotic_instruments.py [hunk 1]
```diff
@@ -25,9 +27,18 @@ def get(self):
...
+                    # and pass them to the frontend_render_info method
+                    # to avoid having to run that query for each instrument
+                    galaxy_catalog_names = session.scalars(
+                        sa.select(Galaxy.catalog_name).distinct()
+                    ).all()
                     retval = {
                         i.id: i.api_class_obsplan.frontend_render_info(
-                            i, self.current_user
+                            i,
+                            self.current_user,
```
Comment (mcoughlin) [inline skyportal/handlers/api/internal/robotic_instruments.py:40]: Any reason this should also be user_or_token?

**Review by mcoughlin — APPROVED** [review]
Looks neat, thanks @Theodlz

File: skyportal/handlers/api/internal/robotic_instruments.py (hunk 1 above)
Comment (Theodlz) [reply skyportal/handlers/api/internal/robotic_instruments.py:40]: Ok it should you are right, I didn't change anything here. We're just lagging behind but nice catch. Let me fix it here and in a couple more places

Comment (Theodlz) [reply skyportal/handlers/api/internal/robotic_instruments.py]: Done @mcoughlin

**Pull request merged**